authenticity.cpp

```diff
@@ -189,37 +185,52 @@ std::string Authenticity::Assay
         }
 
     // Validate all data files.
-    fs::path original_path(fs::current_path());
-    if(0 != chdir(data_path.string().c_str()))
-        {
-        oss
-            << "Unable to change directory to '"
-            << data_path
-            << "'. Try reinstalling."
-            ;
-        return oss.str();
-        }
+    std::vector<md5sum_for_file> sums;
+    std::string md5;
+    std::string md5sums_path = (data_path / md5sum_file()).string();
+
     try
         {
-        system_command("md5sum --check --status " + std::string(md5sum_file()));
+        sums = md5sum_vec_from_file(md5sums_path);
         }
     catch(...)
         {
         report_exception();
         oss
-            << "At least one required file is missing, altered, or invalid."
-            << " Try reinstalling."
+            << "Error processing MD5 sums file '"
```

Comment:
I'm actually not sure if we want to give precise error message here because this is a (trivial) protection mechanism. Could you please check what happens if the checksums don't match with the original version and the new one? I think we should continue giving "At least one required file is missing, altered, or invalid" error if anything goes wrong, as we did before.

Note that this also means we could move try/catch outside all this code and write it only once.